lib/template.html

```diff
@@ -206,28 +209,57 @@
     "headline": {{ seo_page_title | jsonify }},
 {% endif %}
 
+{% if seo_author_name %}
+    "author": {{ seo_author_name | jsonify }},
+{% endif %}
+
 {% if seo_page_image %}
-    "image": {{ seo_page_image | jsonify }},
+    {% if page.image.height && page.image.width %}
+      "image": {
+          "@type": "ImageObject",
+          "url": {{ seo_page_image | jsonify }},
+          "height": {{ page.image.height | jsonify }},
+          "width": {{ page.image.width | jsonify }}
+      },
+    {% else %}
+        "image": {{ seo_page_image | jsonify }},
+    {% endif %}
 {% endif %}
 
 {% if page.date %}
     "datePublished": {{ page.date | date_to_xmlschema | jsonify }},
 {% endif %}
 
+{% if page.dateModified %}
+    "dateModified": {{ page.dateModified | date_to_xmlschema | jsonify }},
+{% elseif page.date %}
+    "dateModified": {{ page.date | date_to_xmlschema | jsonify }},
+{% endif %}
+
 {% if seo_description %}
     "description": {{ seo_description | jsonify }},
 {% endif %}
 
 {% if seo_site_logo %}
     "publisher": {
         "@type": "Organization",
+        {% if seo_author_name %}
+            "name": {{ seo_author_name | jsonify }},
+        {% endif %}
         "logo": {
             "@type": "ImageObject",
             "url": {{ seo_site_logo | jsonify }}
         }
     },
 {% endif %}
 
+{% if seo_type == "BlogPosting" %}
+    "mainEntityOfPage": {
+         "@type": "WebPage",
+         "@id": {{ page.url | replace:'/index.html','/' | absolute_url | jsonify }}
+    },
+{% endif %}
+
```

Comment:
I tried read that post 6 six times, studied Schema.org and still couldn't understand this until I saw your example. 😂 Given Schema.org says all webpages are implicitly of Type WebPage (if not defined) this makes sense. Given this tag also applies to "CreativeWork", it would be useful to add that here while we're making this change as it will help with Structured Data validation for that common type.